pkg/target/target.go

```diff
@@ -257,6 +257,10 @@ matches_namespace_selector(match, ns) {
 `
 
 var libTempl = template.Must(template.New("library").Parse(templSrc))
+var supportedEnforcementActions = []string{
```

Comment:
This should go. The target has no vested interest in which enforcement actions are supported. The set of enforcement points used provides the entire set of actions understood. If enforcement points subscribe to verbs they understand and ignore those they don't, this should work. In the future we should be able to address specific actions to specific enforcement points (e.g. if it's GK dry run, if it's CI/CD, reject).

It's up to the enforcement point to handle known and unknown enforcement actions appropriately.

Reply:
> It's up to the enforcement point to handle known and unknown enforcement actions appropriately.

This logic is within the Gatekeeper enforcement point, it checks for known and unknown enforcement actions within Gatekeeper, currently set to `["DENY", "DRYRUN"]`. IMO I think validating known and unknown supported enforcementAction helps with user experience. Otherwise, they won't know what the supported actions are.

Reply:
We discussed this a bit on the weekly call today. If we are anticipating multiple controllers in the same cluster that can evaluate these constraints, then we should let it pass thru. Otherwise, ideally we do want to validate the user input against a list of supported enforcement actions. The net effect of not validating the values today is any string that is not "deny" will be treated as "dryrun".

cc @teq0 @timothyhinrichs

Reply:
Ah. I didn't realise you meant this to be extracted from GK at some point.

Totally agree with `null -> default deny`, the issue is whether `<unknown value> -> no action` is going to cause scratching of heads and gnashing of teeth. Validation doesn't necessarily tell the user what the possible values might be, but it gives them a hint that they probably should go and check if they've specified something that isn't valid.

But I'm happy to wait for kubectl to support warnings, and stick with `<unknown value> -> no action` + warning when available. I assume that the K8s API will return the warnings as well, for other tooling.

Reply:
Same boat as Craig here.  This being in the GK repo it seemed to make sense to provide as much user feedback as we can.  If this is the same code being run on K8s and in the CI cluster it makes sense to not validate, but then if it's the same code how would there be other supported actions?  It seems the mapping of code to use-cases isn't clear in my head.

Reply:
> Why would the net effect of not validating values today be deny or dry run?

If we are not validate user input for `enforcementAction`, user can enter whatever value they want. We currently only check the following in policy.go. 
https://github.com/open-policy-agent/gatekeeper/blob/ceacda6f194cf7bc5ed8d7423c57da1e5138027e/pkg/webhook/policy.go#L174-L186

So the net effect for `<unknown value>` is `allow` for admission requests and corresponding `status` `violations` will be `enforcementAction:  <unknown value>`. 

So if users happen to type in `DENY` instead of the expected value `deny`, then all requests will be `allow` and users will not know there's an issue until much later.

Reply:
Yeah, it's up for the EP to read and understand the action, the framework just knows how to surface them.

Given that this file is not GK specific, this list should live elsewhere. Maybe in the webhook?